pkg/plugin/manager.go

```diff
@@ -156,9 +156,31 @@ func (m *Manager) Delete(correlationID string) error {
 		logger.Warnf("stopping plugin before delete %s: %s\n", correlationID, err)
 	}
 
-	err = os.RemoveAll(filepath.Dir(plgn.BinPath))
-	if err != nil {
-		return fmt.Errorf("deleting plugin %s: %w", correlationID, err)
+	pluginDir := filepath.Dir(plgn.BinPath)
+
+	alias := Alias(plgn.info.Author, plgn.info.Name)
+	//nolint:nestif
+	if alias == "massa-labs/massa-wallet" {
+		// This is a hack to avoid deleting wallets files of massa wallet plugin
+		// remove all files in the plugin directory except the wallet files starting by wallet_ and ending by .yaml
+		files, err := os.ReadDir(pluginDir)
+		if err != nil {
+			return fmt.Errorf("reading plugin directory %s: %w", pluginDir, err)
+		}
+
+		for _, file := range files {
+			if !(strings.HasPrefix(file.Name(), "wallet_") && strings.HasSuffix(file.Name(), ".yaml")) {
+				err = os.RemoveAll(filepath.Join(pluginDir, file.Name()))
+				if err != nil {
+					return fmt.Errorf("deleting file %s: %w", file.Name(), err)
+				}
+			}
+		}
+	} else {
+		err = os.RemoveAll(pluginDir)
+		if err != nil {
+			return fmt.Errorf("deleting plugin %s: %w", correlationID, err)
+		}
 	}
```

Comment:
You could improve your code by:
- changing the order of your if and return earlier
- put all that in a removePlugin function
- unit test the removePlugin function

My suggestion:
```suggestion
	err = removePlugin(plgn)
	if err != nil {
		return fmt.Errorf("deleting plugin %s: %w", correlationID, err)
	} 
	
```

With the following function:
```golang
// removePlugin removes all the plugin files from the file system.
// However, for the "massa-labs/massa-wallet" plugin, it keeps the wallet files.
// The wallet files are identified by the prefix "wallet_" and the suffix ".yaml".
func removePlugin(plugin *Plugin) error {
	dir := filepath.Dir(plugin.BinPath)

	alias := Alias(plugin.info.Author, plugin.info.Name)
	if alias != "massa-labs/massa-wallet" {
		return os.RemoveAll(dir)
	}

	// for the massa-wallet we want to keep all the wallet entries
	entries, err := os.ReadDir(dir)
	if err != nil {
		return err
	}

	for _, entry := range entries {
		if !(strings.HasPrefix(entry.Name(), "wallet_") && strings.HasSuffix(entry.Name(), ".yaml")) {
			err = os.RemoveAll(filepath.Join(dir, entry.Name()))
			if err != nil {
				return err
			}
		}
	}

	return nil
}
```
The matching unit test:
```golang
func Test_removePlugin(t *testing.T) {
	testCases := []struct {
		name        string
		plugin      *Plugin
		createFiles []string
		remainFiles []string
	}{
		{
			name: "massa-labs/massa-wallet",
			plugin: &Plugin{
				info: &Information{
					Author: "massa-labs",
					Name:   "massa-wallet",
				},
			},
			createFiles: []string{"wallet_test.yaml", "other_file.txt"},
			remainFiles: []string{"wallet_test.yaml"},
		},
		{
			name: "massa-labs/not-massa-wallet",
			plugin: &Plugin{
				info: &Information{
					Author: "massa-labs",
					Name:   "massa-wallet",
				},
			},
			createFiles: []string{"wallet_test.yaml", "other_file.txt"},
			remainFiles: []string{},
		},
		{
			name: "not-massa-labs/massa-wallet",
			plugin: &Plugin{
				info: &Information{
					Author: "massa-labs",
					Name:   "massa-wallet",
				},
			},
			createFiles: []string{"wallet_test.yaml", "other_file.txt"},
			remainFiles: []string{},
		},
	}

	for _, tc := range testCases {
		t.Run(tc.name, func(t *testing.T) {
			// Create a temporary directory for testing.
			tempDir, err := ioutil.TempDir("", "plugin_test")
			if err != nil {
				t.Fatalf("Failed to create temp directory: %v", err)
			}

			tc.plugin.BinPath = filepath.Join(tempDir, "pluginBinary")

			// Create dummy files.
			for _, file := range tc.createFiles {
				dummyFile := filepath.Join(tempDir, file)
				if err := ioutil.WriteFile(dummyFile, []byte("test"), 0666); err != nil {
					t.Fatalf("Failed to create dummy file: %v", err)
				}
			}

			// Call the function to test.
			if err := removePlugin(tc.plugin); err != nil {
				t.Fatalf("Failed to remove plugin: %v", err)
			}

			// Check if the remaining files still exist.
			for _, file := range tc.remainFiles {
				dummyFile := filepath.Join(tempDir, file)
				if _, err := os.Stat(dummyFile); os.IsNotExist(err) {
					t.Fatalf("File was deleted: %v", err)
				}
			}

			// Clean up the temporary directory.
			os.RemoveAll(tempDir)
		})
	}
}
```